polygeist/test/polygeist-opt/sycl/matrix_multiply_reduction.mlir

```diff
@@ -1,4 +1,4 @@
-// RUN: polygeist-opt -licm -raise-scf-to-affine -detect-reduction %s | FileCheck %s
+// RUN: polygeist-opt -arg-promotion -licm -raise-scf-to-affine -detect-reduction %s | FileCheck %s
```

Comment:
After fixing the problem, some operations are unable to be hoisted. By splitting the memref struct argument, they can then be hoisted.

Reply:
I think we need to understand what operation prevent LICM to hoist code out of the loop (in the absence of arg-promotion). I am just worry that arg-promotion hides another problem.

Reply:
Answered in https://github.com/intel/llvm/pull/8872/files#r1153668918.